java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/wiki/commonswiki/CommonsWikimediaPreparation.java

```diff
@@ -346,6 +353,28 @@ private void parseMeta() {
 		}
 	}
 
+	public static String normalizeLang(String jsonStr) {
+		if (jsonStr == null || jsonStr.trim().isEmpty()) {
+			return "{}";
+		}
+		try {
+			Map<String, String> parsed = gson.fromJson(jsonStr, mapType);
+			if (parsed == null || parsed.isEmpty()) {
+				return "{}";
+			}
+			Map<String, String> normalized = new HashMap<>();
+			for (Map.Entry<String, String> e : parsed.entrySet()) {
+				String wikiLangCode = e.getKey().trim();
+				String bcp47 = WikiLangConverter.toBcp47FromWiki(wikiLangCode);
+				normalized.put(bcp47, e.getValue());
```

Comment:
The main issue with the converter is that if we fail to decode a language but return the old result, we end up putting it into the normalized map, even it's not valid. We should not add unparsed languages to the result